### PR TITLE
FOGL-9304 Add upper limit to maximum send latency in south service

### PR DESCRIPTION
**Pull request by MarkRiddoch**

### CODE REVIEW

File: C/services/south/south.cpp [hunk 1]
```diff
@@ -1185,6 +1185,8 @@ void SouthService::addConfigDefaults(DefaultConfigCategory& defaultConfig)
 		}
 	}
 
+	defaultConfig.defaultConfig.setItemAttribute("maxSendLatency", ConfigCategory::MAXIMUM_ATTR, MAXSENLATENCY);
```
Comment (ashish-jabble) [inline C/services/south/south.cpp:1188]: `defaultConfig.defaultConfig` => defaultConfig

**Review by ashish-jabble — CHANGES_REQUESTED** [review]
make compilation is failing, below are the logs
```
17:51:54 make[3]: Leaving directory '/home/ubuntu/Fledge_PR/cmake_build'
17:51:54 CMakeFiles/Makefile2:791: recipe for target 'C/services/south/CMakeFiles/fledge.services.south.dir/all' failed
17:51:54 make[2]: *** [C/services/south/CMakeFiles/fledge.services.south.dir/all] Error 2
17:51:54 make[2]: *** Waiting for unfinished jobs....
17:51:54 [ 66%] Building CXX object C/services/north/CMakeFiles/fledge.services.north.dir/north_plugin.cpp.o
17:52:04 [ 67%] Linking CXX executable fledge.services.north
17:52:04 make[3]: Leaving directory '/home/ubuntu/Fledge_PR/cmake_build'
17:52:04 [ 67%] Built target fledge.services.north
17:52:04 make[2]: Leaving directory '/home/ubuntu/Fledge_PR/cmake_build'
17:52:04 Makefile:135: recipe for target 'all' failed
17:52:04 make[1]: *** [all] Error 2
17:52:04 make[1]: Leaving directory '/home/ubuntu/Fledge_PR/cmake_build'
17:52:04 Makefile:254: recipe for target 'c_build' failed
17:52:04 make: *** [c_build] Error 2
```

File: docs/tuning_fledge.rst [hunk 2]
```diff
@@ -263,6 +263,7 @@ In these circumstances, it is recommended to disable or severely limit the buffe
 
    Data arrives at the buffering subsystem **after** it has passed through the processing pipeline in the south service. Therefore if the pipeline does data compression, for example using the delta filter, this may reduce the arrival rate of data at the buffering subsystem and convert high bandwidth data from the plugin to low bandwidth data to send to the storage subsystem.
 
+The system imposes an upper limit of 600000 milliseconds (10 minutes) on the maximum send latency to prevent it being set so high that it appears that the south service is no longer functioning. This is really only an issue in situationis where the south service does not receive high rates of data and the send latency is set very high. In these cases the data may reside in the south service for a long period, during which it is not accessible to other services within the system. There is also a risk, in these circumstances, that data for a long period of time might be lost if there was a failure that caused the south service to terminate before sending the data to the storage service.
```
Comment (ashish-jabble) [inline docs/tuning_fledge.rst:266]: situationis => situation's

Comment (ashish-jabble) [inline docs/tuning_fledge.rst:266]: It would be even better if we highlight as a note

Comment (MarkRiddoch) [reply docs/tuning_fledge.rst:266]: I didn't want two notes net to each other without any plain text betwen them.

File: docs/tuning_fledge.rst [hunk 3]
```diff
@@ -251,6 +251,10 @@ Reading Latency
 
 Closely related to buffer usage is reading latency in the south service. This is a measure of the delay between the south service receiving a new reading and that reading appearing in the storage subsystem. We deliberately delay the forwarding of readings from the south service to storage in order to create blocks of multiple readings to send per call to the storage layer. This increases the overall throughput of the south to storage interface at the cost of increasing the latency. There are two settings that come into play when defining this, the maximum latency we will accept and the maximum number of readings we will buffer.
 
+.. note::
+
+   The maximum reading latency may be set to any value between 0 and 600000 millisconds. A value of zero will disable the buffering. See below for a discussion of the impact of large values of maximum reading latency.
```
Comment (ashish-jabble) [inline docs/tuning_fledge.rst:256]: milliseconds

**Review by ashish-jabble — APPROVED** [review]

**Pull request merged**